statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitor.java

```diff
@@ -244,7 +244,7 @@ protected String constructMetricName(String finalName) {
 
     protected String constructMetricName(Optional<String> finalName) {
         List<String> copy = new ArrayList<>(names);
-        finalName.ifPresent(copy::add);
+        finalName.map(this::sanitizeNames).ifPresent(copy::add);
```

Comment:
It's strange for me to sanitize something what is called **final**Name.

Reply:
But I can see that we would replace a dot so it's fine for me 👍

Reply:
Even though I think that it might lead to confusion if dots are used for both namespace separators and in regular metric name, I can change implementation in a way that it allows dots. Would that be OK to you?

Reply:
Yep, that would be absolutely OK for me, thanks! 👍